mapie/calibrators/ccp/gaussian.py

```diff
@@ -0,0 +1,338 @@
+        self.multipliers = multipliers
+        self.reg_param = reg_param
+
+    def _check_random_sigma(self) -> bool:
+        """
+        Check ``random_sigma``
+
+        Returns
+        -------
+        bool
+            checked ``random_sigma``
+        """
+        if self.random_sigma is None:
+            return False
+        else:
+            return self.random_sigma
+
+    def _check_points_sigma(
+        self, points: ArrayLike, sigmas: ArrayLike
+    ) -> None:
+        """
+        Take 2D arrays of points and standard deviations and check
+        compatibility
+
+        Parameters
+        ----------
+        points : ArrayLike
+            2D array of shape (n_points, n_in)
+        sigmas : ArrayLike
+            2D array of shape (n_points, 1) or (n_points, n_in)
+
+        Raises
+        ------
+        ValueError
+            If ``sigmas``is not of shape (n_points, 1) or (n_points, n_in)
+        """
+        if _num_samples(points) != _num_samples(sigmas):
+            raise ValueError("There should have as many points as "
+                             "standard deviation values")
+
+        if len(_safe_indexing(sigmas, 0)) not in [
+            1, len(_safe_indexing(points, 0))
+        ]:
+            raise ValueError("The standard deviation 2D array should be of "
+                             "shape (n_points, 1) or (n_points, n_in).\n"
+                             f"Got sigma of shape: ({_num_samples(sigmas)}, "
+                             f"{len(_safe_indexing(points, 0))}).")
+
+    def _check_fit_parameters(
+        self,
+        X: ArrayLike,
+        y_pred: Optional[ArrayLike] = None,
+        z: Optional[ArrayLike] = None,
+    ) -> None:
+        """
+        Check fit parameters. In particular, check that the ``functions``
+        attribute is valid and set the ``functions_``.
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y: ArrayLike of shape (n_samples,)
+            Training labels.
+
+            By default ``None``
+
+        z: Optional[ArrayLike] of shape (n_calib_samples, n_exog_features)
+            Exogenous variables
+
+            By default ``None``
+        """
+        self.random_sigma = self._check_random_sigma()
+        self.points_ = sample_points(X, self.points)
+        self.sigmas_ = compute_sigma(X, self.points, self.points_,
+                                     self.sigma, self.random_sigma)
+        self._check_points_sigma(self.points_, self.sigmas_)
+
+        functions = [
+            lambda X, mu=_safe_indexing(self.points_, i),
+            sigma=_safe_indexing(self.sigmas_, i):
+            np.exp(-0.5 * np.sum(((X - mu) / sigma) ** 2, axis=1))
+            for i in range(_num_samples(self.points_))
+        ]
+        self.functions_ = format_functions(functions, self.bias)
+
+
+def check_calibrator(
+    calibrator: Optional[BaseCalibrator],
+) -> BaseCalibrator:
```

Comment:
Should be move into utils file (even if it needs to call this class). Hoping there will be no circular imports